File: 5-helo/helo.clj
#!/usr/bin/env -S -- clojure -M

(import '[java.lang ProcessBuilder ProcessBuilder$Redirect])
(require '[clojure.java.io :as io])

(->> *file*
     io/file .getName
     (str "HELO :: VIA -- ")
     println)

(println "cannot into chdir")

(->
 (ProcessBuilder. ["bat" "--" *file*])
 (.redirectOutput ProcessBuilder$Redirect/INHERIT)
 (.redirectError ProcessBuilder$Redirect/INHERIT)
 .start .waitFor zero? assert)

File: 5-helo/helo.hs
#!/usr/bin/env -S -- runhaskell

import           System.Environment (getProgName)
import           System.FilePath    ((</>))
import           System.Process     (callProcess)
import           Text.Printf        (printf)

main = do
  name <- getProgName
  let path = "5-helo" </> name
  _ <- putStrLn $ printf "HELO :: VIA -- %s" name
  _ <- putStrLn "cannot into chdir"
  callProcess "bat" ["--", path]
